cpr/multiperform.cpp

```diff
@@ -353,6 +353,9 @@ const std::optional<std::vector<Response>> MultiPerform::intercept() {
     }
 
     if (current_interceptor_ != interceptors_.end()) {
+        if (*current_interceptor_ == nullptr) {
+            return std::nullopt;
+        }
```

Comment:
I'm really not happy this this change. Although at first glance this makes the code more robust, it hides an underlying problem.
To me the only way something like this could ever happen is in case somebody appends a null pointer to the interceptor list (or a pointer that gets null).

In case this happens I would expect my code to segfault highlighting my faulty logic instead of hiding it even further.